code/modules/mob/mob_movement.dm

```diff
@@ -168,8 +168,13 @@
 			direct = newdir
 			n = get_step(mob, direct)
 
-	. = mob.SelfMove(n, direct, delay)
-	if(mob.pulling?.face_while_pulling)
+	var/prev_pulling_loc = null
+	if(mob.pulling)
+		prev_pulling_loc = mob.pulling.loc
+
+	. = mob.SelfMove(n, direct, delay) // The actual movement
+
+	if(prev_pulling_loc && mob.pulling?.face_while_pulling && (mob.pulling.loc != prev_pulling_loc))
```

Comment:
The `pulling` check here is still needed even after line 172, since it's possible for the mob to lose their grab in `SelfMove()`.